features/apimgt/org.wso2.carbon.apimgt.admin.feature/src/main/resources/admin/source/src/app/components/GatewayEnvironments/AddEditGWEnvironment.jsx

```diff
@@ -165,14 +211,28 @@ function AddEditMGLabel(props) {
             Alert.error(formErrors);
             return false;
         }
+        const vhostDto = [];
+        vhosts.forEach((vhost) => {
+            vhostDto.push({
```

Comment:
Can't we simply push `vhost` object?

is this vhostDto refers to an object in React state?

Reply:
`vhostDto` refers to the object that is passing to RestAPI and `vhost` contains some additional properties like `isNew`, `key` that is used for rendering purpose. instead of deleting those properties from the object, I created a new object `vhostDto` so adding new properties to the object does not lead to any errors. Rest API is validating for object fields.